src/main/scala/it/pagopa/interop/agreementprocess/common/readmodel/ReadModelQueries.scala

```diff
@@ -0,0 +1,133 @@
+package it.pagopa.interop.agreementprocess.common.readmodel
+
+import it.pagopa.interop.agreementprocess.common.Adapters._
+import it.pagopa.interop.agreementmanagement.model.agreement.PersistentAgreement
+import it.pagopa.interop.commons.cqrs.service.ReadModelService
+import it.pagopa.interop.agreementprocess.model.AgreementState
+import scala.concurrent.{ExecutionContext, Future}
+import org.mongodb.scala.Document
+import org.mongodb.scala.bson.conversions.Bson
+import org.mongodb.scala.model.Aggregates.{`match`, count, lookup, project, sort, unwind, addFields}
+import org.mongodb.scala.model.{Filters, Field}
+import org.mongodb.scala.model.Projections.{computed, fields, include}
+import org.mongodb.scala.model.Sorts.ascending
+import it.pagopa.interop.agreementmanagement.model.persistence.JsonFormats._
+import it.pagopa.interop.catalogmanagement.{model => CatalogManagement}
+
+object ReadModelQueries {
+  def listAgreements(
+    eServicesIds: List[String],
+    consumersIds: List[String],
+    producersIds: List[String],
+    descriptorsIds: List[String],
+    states: List[AgreementState],
+    showOnlyUpgradeable: Boolean,
+    offset: Int,
+    limit: Int
+  )(readModel: ReadModelService)(implicit ec: ExecutionContext): Future[PaginatedResult[PersistentAgreement]] = {
+
+    val query: Bson =
+      listAgreementsFilters(eServicesIds, consumersIds, producersIds, descriptorsIds, states, showOnlyUpgradeable)
+
+    val filterPipeline: Seq[Bson] = Seq(`match`(query), lookup("eservices", "data.eserviceId", "data.id", "eservices"))
+
+    val upgradablePipeline: Seq[Bson] = {
+      if (showOnlyUpgradeable) {
+        Seq(
+          unwind(fieldName = "eservices.data.descriptors"),
+          addFields(
+            Field(
+              "currentDescriptor",
+              Document("""{ $filter: {
+              input: "$eservices.data.descriptors",
+              as: "descr",         
+              cond: {  $eq: ["$$descr.id" , "$data.descriptorId"]}}} }""")
+            )
+          ),
+          unwind(fieldName = "$currentDescriptor"),
+          addFields(
+            Field(
+              "upgradableDescriptor",
+              Document(
+                """{ $filter: {
+              input: "$eservices.data.descriptors",
+              as: "upgradable",         
+              cond: { $and: [
+                      {$gt:[ {"$toInt" : "$$upgradable.version"} , {"$toInt": "$currentDescriptor.version"}]}, 
+                      {$in:["$$upgradable.state", [""" + CatalogManagement.Published + """, """ +
+                  CatalogManagement.Suspended + """]]}
```

Comment:
Aren't quotes required here?
```suggestion
                      {$in:["$$upgradable.state", ['""" + CatalogManagement.Published + """', '""" +
                  CatalogManagement.Suspended + """']]}
```